packages/hash/api/src/graphql/typeDefs/knowledge/page.typedef.ts

```diff
@@ -0,0 +1,232 @@
+import { gql } from "apollo-server-express";
+
+export const knowledgePageTypedef = gql`
+  type KnowledgePage implements KnowledgeEntity {
+    contents: [KnowledgeBlock!]!
+    archived: Boolean
+    summary: String
+    title: String!
+    icon: String
+
+    # ENTITY INTERFACE FIELDS BEGIN #
+    """
+    The id of the entity
+    """
+    entityId: ID!
+    """
+    The specific version of the entity
+    """
+    entityVersionId: String!
+    """
+    The id of the account that owns this entity.
+    """
+    ownedById: ID!
+    """
+    The fixed id of the type this entity is of.
+    """
+    entityTypeId: ID!
+    """
+    The full entity type definition.
+    """
+    entityType: PersistedEntityType!
+    """
+    The linked entities of the entity.
+    """
+    linkedEntities: [KnowledgeEntity!]!
+    """
+    The JSON object containing the entity's properties.
+    """
+    properties: JSONObject!
+    # ENTITY INTERFACE FIELDS END #
+  }
+
+  type KnowledgeEntityRef {
+    accountId: ID!
+    entityId: ID!
+    entityVersion: String!
+  }
+
+  """
+  Insert a block into a page with a corresponding entity.
+  """
+  input KnowledgeInsertBlockAction {
+    """
+    The account ID to create the block and associated entity in.
+    """
+    accountId: ID!
+    """
+    The position in the page to place the block.
+    """
+    position: Int!
+    """
+    The block componentId.
+    """
+    componentId: ID
+    """
+    The block entity to insert into the page. You should not set a componentId
+    if you provide this
+    """
+    existingBlockEntity: ExistingEntity
+    """
+    The entity to associate with the new block
+    """
+    entity: KnowledgeEntityDefinition!
+    """
+    Allows UpdatePageContentsActions to reference entities created in other actions. Also allows callers to updatePageContents to find the entity id created for this definition in the result. See UpdatePageContentsResult.
+    """
+    blockPlaceholderId: ID
+    """
+    Allows UpdatePageContentsActions to reference entities created in other actions. Also allows callers to updatePageContents to find the entity id created for this definition in the result. See UpdatePageContentsResult.
+    """
+    entityPlaceholderId: ID
+  }
+
+  """
+  Remove a block from a page.
+  """
+  input KnowledgeRemoveBlockAction {
```

Comment:
```suggestion
  input RemoveKnowledgeBlockAction {
```

Reply:
Addressed in 021f42e97909338ec507cf74b5757ed426586e6d